Lib/pathlib.py

```diff
@@ -1512,13 +1512,23 @@ class PosixPath(Path, PurePosixPath):
     """
     __slots__ = ()
 
+    def __new__(cls, *args, **kwargs):
+        if kwargs:
+            raise TypeError("keyword arguments provided but ignored")
+        return super().__new__(cls, *args, **kwargs)
+
 class WindowsPath(Path, PureWindowsPath):
     """Path subclass for Windows systems.
 
     On a Windows system, instantiating a Path should return this object.
     """
     __slots__ = ()
 
+    def __new__(cls, *args, **kwargs):
+        if kwargs:
```

Comment:
This weird for me because if you don't have kwargs on ```__new__(cls, *args, **kwargs)```
because this will raise a exception, what are you sending on ```super().__new__(cls, *args, **kwargs)```

Reply:
Not sure I understand you clearly, but if `kwargs` is an empty dictionary, then `if kwargs` will evaluate to `False`, and the exception will not be raised.